pwndbg/commands/windbg.py

```diff
@@ -141,50 +132,18 @@ def dX(size, address, count, to_string=False, repeat=False):
     """
     Traditionally, windbg will display 16 bytes of data per line.
     """
-    values = []
 
-    if repeat:
-        count = dX.last_count
-        address = dX.last_address
-    else:
-        address = int(address) & pwndbg.gdblib.arch.ptrmask
-        count = int(count)
-
-    type = get_type(size)
-
-    for i in range(count):
-        try:
-            gval = pwndbg.gdblib.memory.poi(type, address + i * size)
-            # print(str(gval))
-            values.append(int(gval))
-        except gdb.MemoryError:
-            break
-
-    if not values:
-        print("Could not access the provided address")
-        return
-
-    n_rows = int(math.ceil(count * size / float(16)))
-    row_sz = int(16 / size)
-    rows = [values[i * row_sz : (i + 1) * row_sz] for i in range(n_rows)]
-    lines = []
-
-    # sys.stdout.write(repr(rows) + '\n')
-
-    for i, row in enumerate(rows):
-        if not row:
-            continue
-        line = [enhex(pwndbg.gdblib.arch.ptrsize, address + (i * 16)), "   "]
-        for value in row:
-            line.append(enhex(size, value))
-        lines.append(" ".join(line))
+    lines = list(
+        chain.from_iterable(
+            pwndbg.hexdump.hexdump(
+                data=None, size=size, count=count, address=address, repeat=repeat, dX_call=True
+            )
+        )
+    )
 
-    if not to_string:
+    if not to_string and lines != []:
```

Comment:
nit: the "Pythonic" way to do this is `if not to_string and lines`: https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty

I'd also accept `len(lines) != 0`, just because it'll catch type issues where `lines` is not a list (even though this isn't considered "Pythonic").

Reply:
Thanks for the advice, I will do it now